### ISSUE TITLE
Egypt - AI Not Improving Flax Resources

### ISSUE THREAD

**Issue by cdanger341** [adaneslavo/More-Unique-Components-for-VP#145]
V31-2

Playing a game as Germany with Rome, Spain, Indonesia, Egypt, and Ethiopia in play to see what their new UIs look like and perform. So far it is in the mid-1700s and after exploring the whole map with my Explorer I noticed that Egypt had still yet to improve the Flax resource that spawned due to building their Nilometers. The Flax only spawned on flat plains/grassland/flood plains and a flat forest tile, but on these tiles Egypt had built farms, villages, and a lumber mill rather than plantations.

I started a game as Egypt and rushed to Nilometers to see if I could improve them and could do so without issue. It looks like something is preventing the Egypt AI from recognizing the Flax resources as something it can now build plantations on.

**Comment by theazuresoul** [comment]
It shouldn't be possible to build farms/villages/lumber mills on tiles with resources. If those improvements are on top of flax, chances are that they were built before the flax spawned (unless there's some unusual bug going on). The AI does not generally build over existing improvements, since resource spawning is not a thing that normally happens in the game.

**Comment by cdanger341** [comment]
Here is a pic of the Flax tiles with farms over them. When I went back to take the screengrab I noticed that one of the Flax tiles was finally improved via a plantation. Given that Nilometers unlock in Classical era and it's currently Industrial, Egypt must have built farms where their Flax resources spawned, but then never went back and improved them. Same for the Flax that spawned on a forest.

![8930_20180515174322_1](https://user-images.githubusercontent.com/19951628/40087329-ab40a33a-5867-11e8-8aaa-6c012faa119d.png)
